prime-router/src/main/kotlin/azure/ConditionMapper.kt

```diff
@@ -52,31 +83,53 @@ class ConditionStamper(private val conditionMapper: IConditionMapper) {
     )
 
     /**
-     * Lookup condition codes for an [observation] and add them as custom extensions
+     * Lookup condition codes and member OIDs for an [observation] and add them as custom extensions
      * @param observation the observation that will be stamped
      * @return a [ObservationStampingResult] including stamping success and any mapping failures
      */
     fun stampObservation(observation: Observation): ObservationStampingResult {
+        // Extract codes and filter out empty values
         val codeSourcesMap = observation.getCodeSourcesMap().filterValues { it.isNotEmpty() }
         if (codeSourcesMap.values.flatten().isEmpty()) return ObservationStampingResult(false)
 
+        // Lookup conditions mapped to codes
         val conditionsToCode = conditionMapper.lookupConditions(codeSourcesMap.values.flatten())
+
+        // Map test codes to member OIDs
+        val memberOidMap = conditionMapper.lookupMemberOid(codeSourcesMap.values.flatten())
```

Comment:
Is there a more targeted approach to looking for specific test codes than passing in all codes from the observation and trying to match them up against the lookup table?